src/ixdat/techniques/spectroelectrochemistry.py

```diff
@@ -0,0 +1,257 @@
+        self,
+        V=None,
+        t=None,
+        index=None,
+        V_ref=None,
+        t_ref=None,
+        index_ref=None,
+    ):
+        """Return the delta optical density Spectrum given a point and reference point.
+
+        Provide exactly one of V, t, and index, and at most one of V_ref, t_ref, and
+        index_ref. For V and V_ref to work, the potential in the measurement must be
+        monotonically increasing.
+
+        Args:
+            V (float): The potential at which to get the spectrum.
+            t (float): The time at which to get the spectrum
+            index (int): The index of the spectrum
+            V_ref (float): The potential at which to get the reference spectrum
+            t_ref (float): The time at which to get the reference spectrum
+            index_ref (int): The index of the reference spectrum
+        Return Spectrum: The dOD spectrum. The data is (spectrum.x, spectrum.y)
+        """
+        if V_ref or t_ref or index_ref:
+            spectrum_ref = self.get_spectrum(V=V_ref, t=t_ref, index=index_ref)
+        else:
+            spectrum_ref = self.reference_spectrum
+        spectrum = self.get_spectrum(V=V, t=t, index=index)
+        field = Field(
+            data=-np.log10(spectrum.y / spectrum_ref.y),
+            name="$\Delta$ OD",
+            unit_name="",
+            axes_series=[self.wavelength],
+        )
+        return Spectrum.from_field(field)
+
+    def track_wavelength(self, wl, width=10, V_ref=None, t_ref=None, index_ref=None):
+        """Return and cache a ValueSeries for the dOD for a specific wavelength.
+
+        The cacheing adds wl_str to the SECMeasurement's data series, where
+            wl_str = "w" + int(wl)
+            This is dOD. The raw is also added as wl_str + "_raw".
+        So, to get the raw counts for a specific wavelength, call this function as
+            and then use __getitem__, as in: sec_meas[wl_str + "_raw"]
+        If V_ref, t_ref, or index_ref are provided, they specify what to reference dOD
+            to. Otherwise, dOD is referenced to the SECMeasurement's reference_spectrum.
+
+        Args:
+            wl (float): The wavelength to track in [nm]
+            width (float): The width around wl to average. For example, if wl=400 and
+                width = 20, the spectra will be averaged between 390 and 410 nm to get
+                the values. Defaults to 10
+            V_ref (float): The potential at which to get the reference spectrum
+            t_ref (float): The time at which to get the reference spectrum
+            index_ref (int): The index of the reference spectrum
+        Returns ValueSeries: The dOD value of the spectrum at wl.
+        """
+        if V_ref or t_ref or index_ref:
+            spectrum_ref = self.get_spectrum(V=V_ref, t=t_ref, index=index_ref)
+        else:
+            spectrum_ref = self.reference_spectrum
+        x = self.wl
+        if width:  # averaging
```

Comment:
Can't understand here, if width is given, then average according to with, else, we average according the default width 10. why here else:  interpolation?

Reply:
Oh, right. The user would have to write `width=False` or `width=0` to force it to interpolate. I have added this now to the docstring.